src/rest/rest_utils.c

```diff
@@ -47,33 +404,23 @@ int coap_to_http_status(int status)
 
 void database_free_entry(database_entry_t *device_entry)
 {
-
     if (device_entry)
     {
-        if (device_entry->uuid)
-        {
-            free(device_entry->uuid);
-        }
-        if (device_entry->psk)
-        {
-            free(device_entry->psk);
-        }
-        if (device_entry->psk_id)
-        {
-            free(device_entry->psk_id);
-        }
+        free(device_entry->uuid);
+        free(device_entry->name);
+        free(device_entry->public_key);
+        free(device_entry->secret_key);
 
         free(device_entry);
     }
 }
 
-int database_validate_new_entry(json_t *j_new_device_object)
+int database_validate_new_entry(json_t *j_new_device_object, linked_list_t *device_list)
```

Comment:
API change is introduced just to see if device in already present. IMO abstract functions that do everything should be avoided. One function check (input) data integrity and other do some other logic, i. e. processing a request in steps:
```
if (database_validate_new_entry(dev) != 0)
    return HTTP_BAD_REQUEST;

if (database_get_by_name(db, dev->name) != NULL)
    return HTTP_BAD_REQUEST;
```